src/nlp/utils/py_utils.py

```diff
@@ -90,8 +90,10 @@ def temporary_assignment(obj, attr, value):
     """Temporarily assign obj.attr to value."""
     original = getattr(obj, attr, None)
     setattr(obj, attr, value)
-    yield
-    setattr(obj, attr, original)
+    try:
+        yield
+    finally:
```

Comment:
This one was pretty bad too, as the cache dir was not renamed to its real name if the `download_from_hf_gcs` had failed. This should be good now